ob-sql-mode.el

```diff
@@ -211,26 +211,18 @@ parameters to the code block.")
   "Return the comint buffer for this session.
 
 Determines the buffer from values in PROCESSED-PARAMS."
-  (let* ((bufname (org-babel-sql-mode--buffer-name processed-params))
-         (sql-bufname (format "*SQL: %s*" bufname))
-         (buf (get-buffer sql-bufname))
-         (product (intern (cdr (assoc :product processed-params)))))
-    (unless (assoc product sql-product-alist)
-      (user-error "Product `%s' is not in `sql-product-alist'" product))
-    (save-current-buffer
-      (unless (sql-buffer-live-p buf product)
-        (if (y-or-n-p (format "Interpreter not running in %s.  Start it? "
-                              sql-bufname))
-            ;; Temporarily redefine pop-to-buffer to do nothing, so
-            ;; that when sql-product-interactive calls it nothing
-            ;; happens.  Otherwise the frame is split to show the
-            ;; interactive buffer, which is not wanted.
-            (let ((old-pop-to-buffer (symbol-function 'pop-to-buffer)))
-              (fset 'pop-to-buffer #'(lambda (&rest r)))
-              (sql-product-interactive product bufname)
-              (fset 'pop-to-buffer old-pop-to-buffer))
-          (user-error "Can't do anything without an SQL interactive buffer")))
-      (get-buffer sql-bufname))))
+  (let* ((sql-product (org-babel-sql-mode-as-symbol (cdr (assoc :product processed-params))))
+         (default-buffer (sql-find-sqli-buffer)))
```

Comment:
> My concern is that it would make the results less repeatable &#x2013; by using the default buffer you're at the mercy of whatever other commands might already have been run in there.

Hi, I undertand your concerns, but I think is a useful addition. It will only use `*SQL*` when **there is not `:product` defined a and there is a `*SQL*`** buffer available, will use that buffer. e.g:

    #+BEGIN_SRC sql-mode
    select 3;
    #+END_SRC

But when you have a `sql-mode` block with `:product`, It will behave as expected: it will ask to start a `postgres` session is there none available.

    #+BEGIN_SRC sql-mode :product postgres
    SELECT count(*) FROM pg_stat_activity;
    #+END_SRC

> I have a similar concern in re prompting the user for the sql-product to use. I think either a default that the user sets, or something that's made explicit in the .org file with a property is the safest way to protect against inadvertent mistakes.

It will only prompt the user when there **not a `:product` defined and there is not a `*SQL*` buffer available**.

My personal use-case would be when I have a `*SQL*` buffer in a remote host and I want to make queries on that buffer instead of opening a new connection which could be difficult in this case.

I can add a warning in the README about this.